projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/deploy/SupportedPythonVersionsTest.java

```diff
@@ -0,0 +1,167 @@
+    ReflectionTestUtils.setField(
+        supportedPythonVersions, "supportedPythonVersions", new HashMap<>());
+
+    Assertions.assertFalse(supportedPythonVersions.isPythonVersionSupported("3.7"));
+  }
+
+  @Test
+  public void isPythonVersionSupported_versionSupported() {
+    var supportedVersions = new HashMap<String, HashMap<String, String>>();
+    supportedVersions.put(
+        "3.7",
+        new HashMap<String, String>() {
+          {
+            put(BASE_IMAGE, "python:3.7-slim");
+            put(VDK_IMAGE, "test_vdk_image");
+          }
+        });
+
+    ReflectionTestUtils.setField(
+        supportedPythonVersions, "supportedPythonVersions", supportedVersions);
+
+    Assertions.assertTrue(supportedPythonVersions.isPythonVersionSupported("3.7"));
+  }
+
+  @Test
+  public void isPythonVersionSupported_versionNotInSupported() {
+    var supportedVersions = new HashMap<String, HashMap<String, String>>();
+    supportedVersions.put(
+        "3.8",
+        new HashMap<String, String>() {
+          {
+            put(BASE_IMAGE, "python:3.8-slim");
+            put(VDK_IMAGE, "test_vdk_image");
+          }
+        });
+
+    ReflectionTestUtils.setField(
+        supportedPythonVersions, "supportedPythonVersions", supportedVersions);
+
+    Assertions.assertFalse(supportedPythonVersions.isPythonVersionSupported("3.7"));
+  }
+
+  @Test
+  public void getSupportedPythonVersions_multipleSupportedVersions() {
+    var supportedVersions = generateSupportedPythonVersionsConf();
+
+    var res = new ArrayList<String>();
+    res.add("3.7");
+    res.add("3.8");
+    res.add("3.9");
+
+    ReflectionTestUtils.setField(
+        supportedPythonVersions, "supportedPythonVersions", supportedVersions);
+
+    Assertions.assertEquals(res, supportedPythonVersions.getSupportedPythonVersions());
+  }
+
+  @Test
+  public void getSupportedPythonVersions_getDefaultVersion() {
+    ReflectionTestUtils.setField(supportedPythonVersions, "defaultPythonVersion", "3.7");
+    var res = new ArrayList<String>();
+    res.add("3.7");
+
+    Assertions.assertEquals(res, supportedPythonVersions.getSupportedPythonVersions());
+  }
+
+  @Test
+  public void getJobBaseImage_defaultImage() {
+    var supportedVersions = generateSupportedPythonVersionsConf();
+    ReflectionTestUtils.setField(
+        supportedPythonVersions, "supportedPythonVersions", supportedVersions);
+    ReflectionTestUtils.setField(supportedPythonVersions, "defaultPythonVersion", "3.7");
+    final String defaultBaseImage = "python:3.7-slim";
+
+    Assertions.assertEquals(defaultBaseImage, supportedPythonVersions.getJobBaseImage("3.11"));
+  }
+
+  @Test
+  public void getJobBaseImage_multipleSupportedVersions() {
+    var supportedVersions = generateSupportedPythonVersionsConf();
+
+    final String resultBaseImg = "python:3.8-slim";
+    ReflectionTestUtils.setField(
+        supportedPythonVersions, "supportedPythonVersions", supportedVersions);
```

Comment:
Create a constant for `supportedPythonVersions` and etc.

Reply:
Done, thanks